gorb/gorb_test.go

```diff
@@ -0,0 +1,149 @@
+package gorb
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	log "github.com/Sirupsen/logrus"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/sky-uk/feed/controller"
+)
+
+func TestE2E(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "E2E Suite")
+}
+
+const (
+	instanceIP       = "10.10.0.1"
+	drainImmediately = 0
+	servicesName     = "http-proxy"
+	servicesPort     = "80"
+	backendWeight    = 1000
+	backendMethod    = "dr"
+	vipLoadbalancer  = "127.0.0.1"
+)
+
+type gorbResponsePrimer struct {
+	response   string
+	statusCode int
+}
+
+type gorbRecordedRequest struct {
+	url  *url.URL
+	body *Backend
+}
+
+type gorbHandler struct {
+	responsePrimer  []gorbResponsePrimer
+	recordedRequest []gorbRecordedRequest
+	requestCounter  int
+}
+
+func (h *gorbHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	defer func() { h.requestCounter++ }()
+	recordedRequest := gorbRecordedRequest{url: r.URL, body: &Backend{}}
+	h.recordedRequest = append(h.recordedRequest, recordedRequest)
+
+	log.Info("Recorded requests: ", len(h.recordedRequest))
+	log.Info("Recorded request url: ", r.URL)
+
+	bodyAsBytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	if len(bodyAsBytes) > 0 {
+		if err := json.Unmarshal(bodyAsBytes, &recordedRequest.body); err != nil {
+			panic(err.Error())
+		}
+	}
+
+	if h.responsePrimer[h.requestCounter].statusCode != 0 {
+		w.WriteHeader(h.responsePrimer[h.requestCounter].statusCode)
+	}
+	if h.responsePrimer[h.requestCounter].response != "" {
+		data, _ := json.Marshal(h.responsePrimer[h.requestCounter].response)
+		w.Write(data)
+	}
+
+}
+
+var _ = Describe("Gorb", func() {
+	var (
+		gorb             controller.Updater
+		server           *httptest.Server
+		serverURL        string
+		responsePrimer   []gorbResponsePrimer
+		recordedRequests []gorbRecordedRequest
+		gorbH            *gorbHandler
+	)
+
+	BeforeSuite(func() {
+		responsePrimer = []gorbResponsePrimer{}
+		recordedRequests = []gorbRecordedRequest{}
+		gorbH = &gorbHandler{responsePrimer: responsePrimer, recordedRequest: recordedRequests}
+		server = httptest.NewServer(gorbH)
+
+		serverURL = server.URL
+		log.Info("url ", serverURL)
+		gorb, _ = New(serverURL, instanceIP, drainImmediately, servicesName, servicesPort, backendWeight, backendMethod, vipLoadbalancer)
+	})
+
+	BeforeEach(func() {
+		gorbH.responsePrimer = []gorbResponsePrimer{}
+		gorbH.recordedRequest = []gorbRecordedRequest{}
+		gorbH.requestCounter = 0
+	})
+
+	AfterSuite(func() {
+		server.Close()
+	})
+
+	Describe("Health endpoint", func() {
+		It("should be healthy when status code is 200", func() {
```

Comment:
How do these tests check we're hitting the correct endpoint for the healthcheck?

Reply:
done